csrc/kernels.cu

```diff
@@ -885,9 +885,16 @@ __global__ void kOptimizer32bit2State(T* g, T* p,
   T g_vals[NUM_PER_THREAD];
   T p_vals[NUM_PER_THREAD];
 
+
   float s1_vals[NUM_PER_THREAD];
   float s2_vals[NUM_PER_THREAD];
 
+  // AdEMAMix has an additional state buffer, which we packed
+  // into state1. We need thread-local storage here for these.
+  // TODO: Mark with [[maybe_unused]] after upgrade to min compiler.
+  float s3_vals[NUM_PER_THREAD];
```

Comment:
There's a few extra memory allocations like this to support AdEMAMix. Have not confirmed if the compiler is optimizing these out for instantiations with `OPTIMIZER=ADAM`, but if not, the overhead isn't very much.